core/app/models/spree/adjustment_reason.rb

```diff
@@ -2,7 +2,7 @@
 
 module Spree
   class AdjustmentReason < Spree::Base
-    has_many :adjustments, inverse_of: :adjustment_reason
+    has_many :adjustments, inverse_of: :adjustment_reason, dependent: :restrict_with_exception
```

Comment:
Maybe we should use `restrict_with_error` here, in order to be able to show the Error message to the Admin while editing Adjustment Reasons?